### PR TITLE
feat: add the ability to duplicate a page

### PR DESCRIPTION
**Pull request by neatbyte-vnobis**
## Changes
Resolves issue "Add the ability to duplicate a page"

## How Has This Been Tested?
Manual

## Documentation
None

### CODE REVIEW

**Comment by SvenAlHamad** [comment]
@neatbyte-vnobis  - on this one, two minor adjustments:
1. Instead of having "Copy" as the label in the drop down menu, please say "Duplicate"
2. When a duplicated page is created, please select it so that the duplicated page selected in the left menu and its preview is show in the right preview bar

**Comment by SvenAlHamad** [comment]
This one is good from my end. @Pavel910 - please check how the code implementation looks.

**Review by Pavel910 — CHANGES_REQUESTED** [review]
@neatbyte-vnobis This solution is problematic  because the `settings` addition to the GraphQL queries is hardcoded, but in reality those settings can be extended with plugins, which means that you will lose information by implementing it this way.

It would be far better to move this whole logic to the API side, and add a `duplicatePage` mutation. It will be cleaner, and easier to handle. Plus, other developers would be able to use this new method programmatically. On the React side, you will just need to call that mutation, and when you get the data back, update the Apollo cache as you're already doing now.

**Comment by neatbyte-vnobis** [comment]
@Pavel910 please review fix for your comment

File: packages/api-page-builder/src/graphql/graphql/pages.gql.ts [hunk 1]
```diff
@@ -452,6 +455,31 @@ const createBasePageGraphQL = (): GraphQLSchemaPlugin<PbContext> => {
...
 
+                    duplicatePage: async (_, args: any, context) => {
+                        try {
+                            const page = await context.pageBuilder.getPage(args.id);
+                            const { id: duplicatedPageId } = await context.pageBuilder.createPage(
+                                page.category
+                            );
+
+                            const duplicatedPageData = {
+                                title: `(Copy) ${page.title}`,
```
Comment (adrians5j) [inline packages/api-page-builder/src/graphql/graphql/pages.gql.ts]:
```suggestion
                                title: `${page.title} (Copy) `,
```

File: packages/app-page-builder/src/admin/plugins/pageDetails/header/pageOptionsMenu/PageOptionsMenu.tsx [hunk 2]
```diff
@@ -69,6 +81,45 @@ const PageOptionsMenu: React.FC<PageOptionsMenuProps> = props => {
...
+            await client.mutate({
+                mutation: DUPLICATE_PAGE,
+                variables: { id: page.id },
+                update(cache, { data }) {
+                    if (data.pageBuilder.duplicatePage.error) {
+                        return;
+                    }
+
+                    GQLCache.addPageToListCache(cache, data.pageBuilder.duplicatePage.data);
+                    showSnackbar(`Duplicated "${page.title}".`);
```
Comment (adrians5j) [inline packages/app-page-builder/src/admin/plugins/pageDetails/header/pageOptionsMenu/PageOptionsMenu.tsx]:
```suggestion
                    showSnackbar(`The page "${page.title}" was duplicated successfully.`);
```

**Review by adrians5j — CHANGES_REQUESTED** [review]
A couple of tiny nit picks, but all in all we're good.
Let's just take care of these and we're good to merge.

Thank you!

**Review by adrians5j — APPROVED** [review]

**Pull request merged**